### PR TITLE
Added "Build release zip" GitHub Action

### LINKED ISSUE #167: Add a "Build release zip" GitHub Action
### Is your enhancement related to a problem? Please describe.

A new Action is released: https://github.com/10up/action-wordpress-plugin-build-zip/. This is already set up in a few PRs: [Insert Special Characters](https://github.com/10up/insert-special-characters/pull/174), [Autoshare for Twitter](https://github.com/10up/autoshare-for-twitter/pull/201) and [Simple Podcasting](https://github.com/10up/simple-podcasting/pull/199). As an example, you can see/download the Simple Podcasting zip [here](https://github.com/10up/simple-podcasting/actions/runs/3596557206).

Please go ahead and create the same in this repository.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change
PR Adds "Build release zip" GitHub Action to generate the release zip before the actual release and Release Instructions are updated to test generated zip before shipping a release.

Closes #167

### How to test the Change
Fork the repo, merged this workflow, and tested the ZIP locally.

### Changelog Entry
Added - "Build release zip" GitHub Action.

### Credits

Props @faisal-alvi 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: .github/workflows/build-release-zip.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Install Composer dependencies
+        if: steps.cache-vendor.outputs.cache-hit != 'true'
+        run: composer install --no-dev
+
+      - name: Build plugin
+        run: npm run build
+
+      - name: Generate ZIP file
+        uses: 10up/action-wordpress-plugin-build-zip@stable
```
Comment (peterwilsoncc) [inline .github/workflows/build-release-zip.yml]: Nit: EOF line break would be nice.

File: .github/workflows/build-release-zip.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        id: cache-composer
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-vendor
+        with:
+          path: |
+            vendor
+            ~/.composer/cache
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/composer.json') }}
+
```
Comment (peterwilsoncc) [inline .github/workflows/build-release-zip.yml]: Composer only uses dev dependencies and isn't loaded in production (ie, autoload isn't referenced) so can be ignored from the build here. It's not included in the wp.org version either, see https://plugins.trac.wordpress.org/browser/simple-local-avatars/trunk

**Review by peterwilsoncc — CHANGES_REQUESTED** [review]
I reviewed the zip file generated on your repo in the action https://github.com/faisal-alvi/simple-local-avatars/actions/runs/3603276090

As noted inline, it's including some files that aren't needed.

File: .github/workflows/build-release-zip.yml (hunk 2 above)
Comment (faisal-alvi) [reply .github/workflows/build-release-zip.yml]: The composer dependencies install step is removed.

**Review by peterwilsoncc — APPROVED** [review]
Thank you kindly, this looks good.

Are you able to re-run against your fork and self merge once you confirm the zip matches that on wp.org?

**Comment by faisal-alvi** [comment]
@peterwilsoncc I checked the new [zip](https://github.com/faisal-alvi/simple-local-avatars/actions/runs/3628003182) and it now matches with the wp.org (`vendor` folder is not included), thank you!

**Pull request merged**